site/app/templates/Authentication.twig

```diff
@@ -11,7 +11,7 @@
             {% for key, value in old %}
                 <input type='hidden' name='old_{{ key }}' value='{{ value }}' />
             {% endfor %}
-            <input type="text" name="user_id" placeholder="User ID" required autofocus/><br />
+            <input type="text" name="user_id" onchange="$(this).val($(this).val().trim())" placeholder="User ID" required autofocus/><br />
```

Comment:
@bmcutler @drumilpatel2000 This line checks for spaces only on the left and right of the user id input. One would expect instead to disallow typing in spaces at all. Changing the `onchange` to `onkeyup` would effectively implement the prohibition of typing in whitespace in the userid textbox.